cli/src/pcluster/templates/pcluster_dashboard_builder.py

```diff
@@ -0,0 +1,586 @@
+        self._add_text_widget("\n## Head Node Instance Metrics\n", cloudwatch_dashboard)
+
+        # Create EC2 metrics graph for head node instance
+        ec2_metrics = [
+            {"metrics": ["CPUUtilization"], "title": "CPU Utilization"},
+            {"metrics": ["NetworkPacketsIn", "NetworkPacketsOut"], "title": "Network Packets In/Out"},
+            {"metrics": ["NetworkIn", "NetworkOut"], "title": "Network In and Out"},
+            {"metrics": ["DiskReadBytes", "DiskWriteBytes"], "title": "Disk Read/Write Bytes"},
+            {"metrics": ["DiskReadOps", "DiskWriteOps"], "title": "Disk Read/Write Ops"},
+        ]
+        widgets_list = []
+        for metrics_param in ec2_metrics:
+            metrics_list = self._generate_ec2_metrics_list(metrics_param["metrics"])
+            graph_widget = self._generate_graph_widget(metrics_param["title"], metrics_list)
+            widgets_list.append(graph_widget)
+        cloudwatch_dashboard.add_widgets(*widgets_list)
+        self._update_coord_after_section(self.coord, self.graph_height)
+
+    def _add_ebs_metrics_graphs(self, cloudwatch_dashboard, ebs_metrics, conditional_ebs_metrics):
+        # Create a text widget for subtitle "EBS Metrics"
+        self._add_text_widget("\n## EBS Metrics\n", cloudwatch_dashboard)
+
+        # Create unconditional EBS metrics graph widgets
+
+        # Get a list of shared EBS volumes objects
+        ebs_volumes_list = []  # TODO: Use the list of shared EBS in self.storage_resource
+        for storage in self.config.shared_storage:
+            if storage.shared_storage_type == SharedStorageType.EBS:
+                ebs_volumes_list.append(storage)
+        widgets_list = []
+        widgets_list += self._add_storage_widgets(ebs_metrics, ebs_volumes_list, "AWS/EBS", "VolumeId", "volume_id")
+
+        # Add Conditional EBS metrics
+        widgets_list += self._add_conditional_storage_widgets(
+            conditional_metrics=conditional_ebs_metrics,
+            volumes_list=ebs_volumes_list,
+            namespace="AWS/EBS",
+            dimension_vol_name="VolumeId",
+            attribute_name="volume_id",
+            vol_attribute_name="volume_type",
+        )
+        cloudwatch_dashboard.add_widgets(*widgets_list)
+        self._update_coord_after_section(self.coord, self.graph_height)
+
+    def _add_raid_metrics_graphs(self, cloudwatch_dashboard, ebs_metrics, conditional_ebs_metrics):
+        self._add_text_widget("\n## RAID Metrics\n", cloudwatch_dashboard)
+
+        # Get a list of Raid volumes objects
+        raid_volumes_list = []  # TODO: Use the list of Raid in self.storage_resource
+        for storage in self.config.shared_storage:
+            if storage.shared_storage_type == SharedStorageType.RAID:
+                raid_volumes_list.append(storage)
+
+        widgets_list = []
+        widgets_list += self._add_storage_widgets(ebs_metrics, raid_volumes_list, "AWS/EBS", "VolumeId", "volume_id")
```

Comment:
nit:
it's useless to initialize widgets_list to an empty list and then use += to populate it.
You can directly do: `widgets_list = self._add_storage_widgets`